database/src/main/scala/cromwell/database/slick/JobStoreSlickDatabase.scala

```diff
@@ -10,23 +10,36 @@ trait JobStoreSlickDatabase extends JobStoreSqlDatabase {
 
   import dataAccess.driver.api._
 
-  override def addJobStoreEntries(entries: Iterable[JobStoreEntry])(implicit ec: ExecutionContext): Future[Unit] = {
-    val action = dataAccess.jobStoreAutoInc ++= entries
+  private def addJobStoreEntry(jobStoreEntry: JobStoreEntry, jobStoreSimpletonEntries: Iterable[JobStoreResultSimpletonEntry])
+                              (implicit ec: ExecutionContext): DBIO[Unit] = {
+    for {
+      insertJobStoreEntry <- dataAccess.jobStoreAutoInc += jobStoreEntry
+      entries = jobStoreSimpletonEntries map { _.copy(jobStoreId = insertJobStoreEntry.self) }
```

Comment:
Heh. I feel the same way about scalaz, and I think it's because slick is using monads here too. 

In this case, `SingleInsertResult` is just the auto incremented `jobStoreId`. I also have no clue what/where `.self` is.

This would probably read better as:

```scala
insertedJobStoreId <- dataAccess.jobStoreAutoInc += jobStoreEntry
entries = jobStoreSimpletonEntries map { _.copy(jobStoreId = insertedJobStoreId) }
```